src/library_int53.js

```diff
@@ -83,18 +83,32 @@ mergeInto(LibraryManager.library, {
   // converts it to a JavaScript Number, which can represent 53 integer bits precisely.
   // TODO: Add $readI53FromI64Signaling() variant.
   $readI53FromI64: function(ptr) {
-    return HEAPU32[ptr>>2] + HEAP32[ptr+4>>2] * 4294967296;
+    var result = HEAPU32[ptr>>2] + HEAP32[ptr+4>>2] * 4294967296;
+#if ASSERTIONS
+    var lo = HEAPU32[ptr>>2];
+    var hi = HEAP32[ptr+4>>2];
+    if ((result >>> 0) != (lo >>> 0)) warnOnce('readI53FromI64() resulted in rounding lo=0x' + lo.toString(16) + ', hi=0x' + hi.toString(16) + ' to 0x' + result.toString(16));
+#endif
+    return result;
```

Comment:
Oops, I thought surely it would be able to optimize that. Will do.

Reply:
Hm, actually, this is optimizing just fine in my local build. Here's readI53FromU64 (since that's used by the build I have on hand):

```js
function Q(a) {
  return G(4 * (a >> 2), 4, 1) + 4294967296 * G(4 * (a + 4 >> 2), 4, 1);
}
```